### ISSUE TITLE
Row of tabs on the toolbar

### ISSUE THREAD

**Issue by K4sum1** [fork-maintainers/iceraven-browser#674]
In Fennec, if the display was big enough, you would get a tab bar on top of the toolbar. This is what's been stopping me from using a newer Firefox based browser. (I'm using Vivaldi in the meantime, but I hate the two bar layout) I would like an option to bring them back.

Reference image of Fennec in tablet mode.
https://upload.wikimedia.org/wikipedia/commons/c/ca/Firefox_Android_Tablet_mode.png

**Comment by nakele** [comment]
Same here, patently waiting for this feature to land.

**Comment by akliuxingyuan** [comment]
It's a good feature to have, I find a same request in fenix https://github.com/mozilla-mobile/fenix/issues/2344

**Comment by Barkuti** [comment]
What the hell? The Mozilla morons can't realize Firefox is also used on tablets, Android TVs (where a touch screen is missing) and even by power users like me who change their screen smallest width setting to enter tablet territory? 😒😒😒 Sucks!
Piss poor, really piss poor way of doing things.
The issue is now in Bugzilla: [Desktop like tab bar option](https://bugzilla.mozilla.org/show_bug.cgi?id=1812934).

**Comment by K4sum1** [comment]
> even by power users like me who change their screen smallest width setting to enter tablet territory? 

4.6in screen. I loved doing this and using Fennec F-Droid 68. Now I use Vivaldi, but wish I could use something else designed better.
![Screenshot_20240330-075926952](https://github.com/fork-maintainers/iceraven-browser/assets/77039240/9ad54fda-edd5-4be3-a33e-e5e966e9cddf)

**Comment by Barkuti** [comment]
First of all, I'd prefer to just use or speak of the **dpi** or dots per inch variable or setting, but that's not what can be set on my custom display size menu, and neither in any Android version or ROM which I I know of. All you have is **Display size** in the **Display** menu within settings, which sets the smallest width and dpi in an unrefined, *average Joe* way. Since I am using a good ol' AOSP custom ROM, I can adjust the **Smallest width** setting in my Display menu. I've only seen this feature in my Android 8.1 AOSP 5.7 ROM (from developer Davide, dmd79 @ xdaforums) and it's also on his later Android 9, 10, 11 & 12 releases, if I do recall correctly. But in other custom ROMs or devices it is necessary to enable an use developer options to fine tune the smallest width setting.
The smallest width setting is inversely proportional to the dpi setting. So: smallest_width × dpi = **swdpi**, where swdpi is a constant value.

Having said the above:

[K4sum1](https://github.com/K4sum1), not exactly. The display size menu is of no use to me; even on its lowest setting it is still **far** away.
My device's stock smallest width setting is 360 dp, which multiplied by 480 dpi equals 172800 (= swdpi):

![Screenshot_Settings_20240330-210015](https://github.com/fork-maintainers/iceraven-browser/assets/28562086/f4ae2206-beda-4c6f-a612-2f228843aa1a)

Setting my display size to smallest in the display size menu I attain 514 dp as smallest width (and this way my modified dpi is 172800 / 514 = 336.186770428016 dpi):

![Screenshot_Settings_20240330-230836](https://github.com/fork-maintainers/iceraven-browser/assets/28562086/b4e237fd-88fd-425a-8561-7da5098dbab9)
Since a Zuk Z2 Pro is said to have a screen with 5.2" of diagonal, reducing its dpi to that point would be like if it had 514 / 360 × 5.2" = 7.424444444446… ≈7.42 inches of apparent screen diagonal.

Once I reach 720 dp and up in the smallest width setting, my desktop and menus change to what I call “tablet territory”: the navigation bar starts appearing at the bottom when in landscape (heck, I don't like the navigation bar taking up precious screen height when in landscape but fortunately it can be disabled), Hacker's Keyboard keys appear smaller with increased key distance, and the drop down menu becomes slim, not taking up all the screen width, and appears centered where I use my finger to drop it:

![Screenshot_Settings_20240330-230950](https://github.com/fork-maintainers/iceraven-browser/assets/28562086/e53e4c0b-05c2-4c4f-965e-d16f12b5dee4)
![Screenshot_Settings_20240330-231031](https://github.com/fork-maintainers/iceraven-browser/assets/28562086/a4a137b6-83cb-4e12-8049-69810c916c60)
The second image is with 115% font size, my usual setting; it increases the text size in my device menus without affecting applications.

A 800 dp smallest width setting is more than twice the stock setting, and its corresponding dpi value is 216 (≈11.56“ of apparent screen diagonal vs stock). Check this landscape screenshot at such setting, showing the slim drop down menu with recents in the background:

![Screenshot_IU_del_sistema_20240331-001031](https://github.com/fork-maintainers/iceraven-browser/assets/28562086/666da42d-927e-4ada-850f-2fe9a7928eae)

I love this AOSP ROM I am using so much that, unless I find some other ROM for a different device which at least comes close, it's going to be a @#$% (thank you Davide!).

As a curious and tad off topic note, WhatsApp misbehaves on a first startup (a reinstall or clear app data reinitialization) if my smallest width is set to 600 dp and above, dumbarsely asking me to pair WhatsApp with my device (WTF???) while showing me a QR code on screen, instead of actually asking the Android OS if the device has phone capabilities and acting accordingly or letting the user choose. This is WhatscrApp's fault, because its designers or developers apparently think that if a device's screen is determined to be rather big, it must be a tablet without phone capabilities. Wrong, wrong and wrong WhatscrApp, **wrong**. The solution is easy, of course, and I really wonder if any phone tablet user has ever found this problem…

**Comment by K4sum1** [comment]
I used to do ro.sf.lcd_density 240 in build.prop, but that quit working in either 9 or 10, so I just do it the normie way. I've been wanting to make a non-cursed 8.1 or 7.1.2 build for my phones, but I don't have the skills to do it.

**Comment by Barkuti** [comment]
ro.sf.lcd_density 240 is 240 dpi, doesn't it?
Check developer settings (tap a bunch of times over the compilation number in **About phone**), you should be able to custom set the smallest width there (and also do yourself a favour and disable all time-wasting stupid animations, if you will).

P.S.: to hell with G00gl€ and all of its crappy, **downgraded** newer Android versions. Google talks about security reasons, but the consequences for power users who don't need such bullshite are a @#$%. So, stick this 🖕 up your arse G00gl€, you deserve it.

**Comment by K4sum1** [comment]
Yeah I think smallest scaling for my phone is like 220 or so, but it's fine. I'll just keep using what works for me.

I know a lot more than just disabling animations too. My current ROM is a custom compiled RR Q for my device that I built myself.

**Comment by Barkuti** [comment]
You built a ROM? Do you have development skills?
220 dpi is nice, but can't help to say that it is a senseless limit. Bollocks what I said a minute ago, my device only goes down to 120 dpi (1440 dp).

**Comment by K4sum1** [comment]
> You built a ROM? Do you have development skills?

Kinda sorta. I can't really make code or easily debug issues, but I can figure out things enough to modify something to work.

https://github.com/Eclipse-Community/build_lineage/tree/rr-q

> 220 dpi is nice, but can't help to say that it is a senseless limit.

Ok, but my eyes think otherwise. I don't think my phone would be usable too much below 220. Have to remember, 4.6in 720p screen.

**Comment by akliuxingyuan** [comment]
Introduced in latest version. `settings > customize > tab bar display`
